vuplay.js

```diff
@@ -1,75 +1,128 @@
+    var fairplayLicenseServerUrl = "https://fairplay-license.vudrm.tech/license";
+
+    // Please login to https://admin.vudrm.tech to generate a vudrm token.
     var vudrmToken = "<vudrm-token>";
 
-    // Set polyfills required by shaka
-    shaka.polyfill.installAll();
-    if (shaka.Player.isBrowserSupported()) {
+    // A bit of hacky way to detect Safari but will do for demo purposes...
+    var isSafari = (navigator.userAgent.indexOf("Safari") != -1 && navigator.userAgent.indexOf("Chrome") == -1);
+
+    // Fetch the fairplay certificate used to generate the fairplay license request
+    function getFairplayCertificate() {
+        var certRequest = new XMLHttpRequest();
+        certRequest.responseType = "arraybuffer";
+        certRequest.open("GET", fairplayCertificateUrl, true);
+        certRequest.onload = function (event) {
+            if (event.target.status == 200) {
+                loadPlayer(new Uint8Array(event.target.response));
+            } else {
+                var error = new Error("HTTP status: " + event.target.status + " when getting Fairplay Certificate.");
+                onError(error);
+            }
+        };
+        certRequest.send();
+    }
+
+    // Returns a shaka player config for use with mpeg-dash streams
+    function getNonSafariPlayerConfig() {
+        return {
+            drm: {
+                servers: {
+                    "com.widevine.alpha": "https://widevine-license.vudrm.tech/proxy",
+                    "com.microsoft.playready": "https://playready-license.vudrm.tech/rightsmanager.asmx"
+                }
+            }
+        }
+    }
+
+    // returns a shaka player config for use with HLS streams
+    function getSafariPlayerConfig(fairplayCertificate) {
+        return {
+            drm: {
+                servers: {
+                    "com.apple.fps.1_0": fairplayLicenseServerUrl
+                },
+                advanced: {
+                    "com.apple.fps.1_0": {
+                        serverCertificate: fairplayCertificate
+                    }
+                },
+                initDataTransform: function (initData, initDataType) {
+                    if (initDataType == "skd") {
+                        // Set the Fairplay license server URL with the one from the HLS manifest
+                        fairplayLicenseServerUrl = shaka.util.StringUtils.fromBytesAutoDetect(initData);
+                        
+                        // Create the initData for Fairplay
+                        var contentId = fairplayLicenseServerUrl.split("/").pop();
+                        var certificate = window.shakaPlayerInstance.drmInfo().serverCertificate;
+                        return shaka.util.FairPlayUtils.initDataTransform(initData, contentId, certificate);
+                    } else {
+                        return initData;
+                    }
```

Comment:
What do you think about inverting the if test and transforming it in a quick return?
```suggestion
                    if (initDataType != "skd") { return initData; }

                    // Set the Fairplay license server URL with the one from the HLS manifest
                    fairplayLicenseServerUrl = shaka.util.StringUtils.fromBytesAutoDetect(initData);

                    // Create the initData for Fairplay
                    var contentId = fairplayLicenseServerUrl.split("/").pop();
                    var certificate = window.shakaPlayerInstance.drmInfo().serverCertificate;
                    return shaka.util.FairPlayUtils.initDataTransform(initData, contentId, certificate);
```

Reply:
We can do... Personally I always prefer to test for "truth" and not return early from these things but that is just personal codign style!

Reply:
I personnaly prefer improved readability. and removing a level of indentation make it more readable for me... 😉 

Again only my opinion... 😄